dts/bindings/coredump/zephyr,coredump.yaml

```diff
@@ -0,0 +1,16 @@
+# Copyright Meta Platforms, Inc. and its affiliates.
+# SPDX-License-Identifier: Apache-2.0
+
+include: base.yaml
+
+compatible: "zephyr,coredump"
+
+description: Pseudo-device to help capturing desired data into core dumps
+
+properties:
```

Comment:
I would suggest 2 possible cases for the dt bindings (as enums)
1. `COREDUMP_TYPE_MEMCPY` (array of memory regions would work)
2. `COREDUMP_TYPE_CALLBACK` (desired size of the coredump memory, and other code may register a callback)

E.g.
```
/ {
	coredump_device0: coredump-device0 {
		compatible = "zephyr,coredump";
		label = "coredump_device_0";
		status = "okay";
    type = COREDUMP_TYPE_CALLBACK;
    /* coredump device could statically allocate array of sufficient size (42) 
     */
    memory-regions = <0x00000000 42>;
	};

	other_device1: other-device1 {
		compatible = "other,device";
		label = "other_device_1";
		status = "okay";
		/* this device (or any arbitrary code that opens the 	coredump_device0 handle)
		 * could register a callback for application-specific coredump behaviour */
    coredump-device = &coredump_device0;
	};
};
```